File: Activit_and_Hooks/app/src/main/java/com/example/activit_and_hooks/MainActivity.java
package com.example.activit_and_hooks;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    public void onSecondActivity(View view){
        Log.i("INFO", "SECOND ACTIVITY");
        Intent intent = new Intent(this, SecondActivity.class);
        startActivity(intent);

//        Explicit
    }


    private void onIntentFilter(View view){
//        Implicit
        Intent sendingIntent = new Intent();

        sendingIntent.setAction(Intent.ACTION_SEND);
        sendingIntent.putExtra(SecondActivity.EXTRA_MESSAGE, "IMPLICIT INTENT MESSAGE");
        sendingIntent.setType("text/plain"); // MIME


        Intent windowChooser = Intent.createChooser(sendingIntent, "Chose Application");


        if( sendingIntent.resolveActivity( getPackageManager()) != null ) {
            startActivity(windowChooser);
        }
        else {
            Toast.makeText(this, "windowChooser failed.", Toast.LENGTH_SHORT).show();
        }

    }



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final Button onIntentFilterBtn = (Button) findViewById(R.id.onIntentFilterBtn);
        onIntentFilterBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onIntentFilter(v);
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i("ACTIVITY_LIFECYCLE", "MainActivity.onStart");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i("ACTIVITY_LIFECYCLE", "MainActivity.onRestart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("ACTIVITY_LIFECYCLE", "MainActivity.onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i("ACTIVITY_LIFECYCLE", "MainActivity.onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i("ACTIVITY_LIFECYCLE", "MainActivity.onStop");
    }
}

File: Email_Password_Firebase_G11/app/src/main/java/com/example/email_password_firebase_g11/MainActivity.kt
package com.example.email_password_firebase_g11

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import com.google.firebase.auth.FirebaseAuth

class MainActivity : AppCompatActivity() {
    private lateinit var auth: FirebaseAuth
    private lateinit var singOutButton: Button
    private lateinit var updatePasswordButton: Button
    private lateinit var userDisplayName: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        auth = FirebaseAuth.getInstance()
        singOutButton = findViewById(R.id.signOutButton)
        updatePasswordButton = findViewById(R.id.updatePasswordButton)
        userDisplayName = findViewById(R.id.userDisplayName)

        singOutButton.setOnClickListener {
            auth.signOut()
            val intent = Intent(this, LoginActivity::class.java)
            startActivity(intent)
            finish()
        }

        updatePasswordButton.setOnClickListener {
            val intent = Intent(this, UpdatePasswordActivity::class.java)
            startActivity(intent)
//            finish()
        }

        if( auth.currentUser != null ) {
            val displayName = auth.currentUser?.displayName
            userDisplayName.text = displayName
        }

    }
}